File: PROIECT_II-improvements/cinema/cinema-api/Cinema.API/DataLogic/Entities/Movie.cs
using Microsoft.EntityFrameworkCore;
using Microsoft.EntityFrameworkCore.Metadata.Builders;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DataLogic.Entities
{
    public class Movie
    {
        public Guid Id { get; set; }

        public Guid? PhotoId { get; set; }

        public string Title { get; set; }

        public string Description { get; set; }

        public TimeSpan Duration { get; set; }

        public Photo? Photo { get; set; }

        public ICollection<Genre>? Genres { get; set; }

        public ICollection<Showing>? Showings { get; set; }
    }

    public class MovieConfiguration : IEntityTypeConfiguration<Movie>
    {
        public void Configure(EntityTypeBuilder<Movie> configuration)
        {
            configuration.HasOne(_ => _.Photo)
                .WithOne(_ => _.Movie)
                .HasForeignKey<Movie>(_ => _.PhotoId)
                .OnDelete(DeleteBehavior.NoAction);
        }
    }
}

File: PROIECT_II-improvements/cinema/client/src/app/models/movie.ts
import { Time } from "@angular/common";
import { GenreType } from "../enums/genreType";
import { Showing } from "./showing";

export interface Movie {
    id: string;
    title: string;
    description: string;
    duration: string;
    photoUrl: string;
    genres: { "name": GenreType }[];
    showings: Showing[];
}